src/telemetry/telemetryController.ts

```diff
@@ -58,34 +58,44 @@ type NewConnectionTelemetryEventProperties = {
   is_used_saved_connection: boolean;
 };
 
+type DocumentUpdatedTelemetryEventProperties = {
+  source: string;
+  success: boolean;
+};
+
 export type TelemetryEventProperties =
   | PlaygroundTelemetryEventProperties
   | LinkClickedTelemetryEventProperties
   | ExtensionCommandRunTelemetryEventProperties
-  | NewConnectionTelemetryEventProperties;
+  | NewConnectionTelemetryEventProperties
+  | DocumentUpdatedTelemetryEventProperties;
 
 export enum TelemetryEventTypes {
   PLAYGROUND_CODE_EXECUTED = 'Playground Code Executed',
   EXTENSION_LINK_CLICKED = 'Link Clicked',
   EXTENSION_COMMAND_RUN = 'Command Run',
   NEW_CONNECTION = 'New Connection',
   PLAYGROUND_SAVED = 'Playground Saved',
-  PLAYGROUND_LOADED = 'Playground Loaded'
+  PLAYGROUND_LOADED = 'Playground Loaded',
+  DOCUMENT_UPDATED = 'Document Updated'
 }
 
 /**
  * This controller manages telemetry.
  */
 export default class TelemetryController {
+  public _needTelemetry: boolean;
```

Comment:
Is `public` with the underscore on purpose?

Reply:
No, it should be private. Left over after refactoring. Good catch!